docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -394,6 +396,49 @@ Some tools, such as `autoconf`, automatically determine what the compiler suppor
 
 ---
 
+### Enable warnings for possibly misleading unicode bidirectional control characters
+
+| Compiler Flag                                                             | Supported since | Description                                                                                                                                    |
+| ------------------------------------------------------------------------- | ----------------| ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| <span id="-Wbidi-chars=any,ucn">`-Wbidi-chars=any,ucn`</span>             | GCC 12          | Enable warnings for any UTF-8 and UCN bidirectional control characters in comments, string literals, character constants, and identifiers       |
```

Comment:
Double-check whether the ucn option is relevant for security. 

Need test case for confirming whether the ucn option affects UCNs in general or just bidi UCNs.

Reply:
Based on [this test case](https://godbolt.org/z/McssM61Ye) `-Wbidi-char=any,ucn` only warns on UCNs that correspond to bidi characters in string literals, character constants, and identifiers. Non-bidi UCNs in comments, string literals, character constants, and identifiers are not affected, and neither are bidi UCNs in comments.

Reply:
It doesn't matter for *Trojan* Source, since they're clearly visible in the source code as `\uXXXX` or \U....`. See the patch: https://gcc.gnu.org/pipermail/gcc-patches/2021-November/583031.html